pkg/jx/cmd/common_install.go

```diff
@@ -1138,24 +1150,53 @@ func (o *CommonOptions) installJx(upgrade bool, version string) error {
 	if err != nil {
 		return err
 	}
-	err = util.UnTargz(tarFile, jxHome, []string{binary, fileName})
-	if err != nil {
-		return err
-	}
-	err = os.Remove(tarFile)
-	if err != nil {
-		return err
-	}
-	err = os.Remove(binDir + "/jx")
-	if err != nil && o.Verbose {
-		log.Infof("Skipping removal of old jx binary: %s\n", err)
-	}
-	// Copy over the new binary
-	err = os.Rename(jxHome+"/jx", binDir+"/jx")
-	if err != nil {
-		return err
+
+
+	if runtime.GOOS != "windows" {
+		err = util.UnTargz(tmpArchiveFile, jxHome, []string{binary, fileName})
```

Comment:
this was existing code - it looks wrong to me as it is asking to unpack "binary" and filename" only they will either be the same or only binary will be present as far as I understand this